src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -119937,12 +120682,12 @@ class Program
                 // (7,71): warning CS8601: Possible null reference assignment.
                 //     static void F3<T>(   [AllowNull]T x, [DisallowNull]ref T y) { y = x; } // 3
                 Diagnostic(ErrorCode.WRN_NullReferenceAssignment, "x").WithLocation(7, 71),
-                // (8,71): warning CS8601: Possible null reference assignment.
-                //     static void F4<T>(   [AllowNull]T x,    [MaybeNull]ref T y) { y = x; }
-                Diagnostic(ErrorCode.WRN_NullReferenceAssignment, "x").WithLocation(8, 71),
                 // (9,71): warning CS8601: Possible null reference assignment.
                 //     static void F5<T>(   [AllowNull]T x,      [NotNull]ref T y) { y = x; } // 4
```

Comment:
>T x [](start = 55, length = 3)

I think this is maybe innate to the problem, but it feels odd to have a different message here depending on whether `AllowNull` was used. In F5, T is "definitely maybe null", but in FA, T is "possibly maybe null".

Reply:
We could have the new check (from `CheckDisallowedNullAssignment`) report the same diagnostic as the regular checks that `VisitConversion` can report. But since we have more context/information here, I figured out that some hint would be helpful, when possible. But it is a bit inconsistent, I agree.

---
In reply to: [368176797](https://github.com/dotnet/roslyn/pull/40789#discussion_r368176797) [](ancestors = 368176797)